linter/lints/cabf_br/lint_crl_validity_period.go

```diff
@@ -42,17 +47,66 @@ func (l *crlValidityPeriod) CheckApplies(c *x509.RevocationList) bool {
 }
 
 func (l *crlValidityPeriod) Execute(c *x509.RevocationList) *lint.LintResult {
-	validity := c.NextUpdate.Sub(c.ThisUpdate)
-	if validity <= 0 {
+	/*
+	   Let's Encrypt issues two kinds of CRLs:
+
+	    1) CRLs containing subscriber certificates, created by crl-updater.
+	       These assert the distributionPoint and onlyContainsUserCerts
+	       boolean.
+	    2) CRLs containing issuer CRLs, created by the ceremony tool. These
+	       assert the onlyContainsCACerts boolean.
+
+	   We use the presence of these booleans to determine which BR-mandated
+	   lifetime to enforce.
+	*/
+
+	// The only way to determine which type of CRL we're dealing with. The
+	// issuingDistributionPoint must be parsed and the internal fields
+	// inspected.
+	idpOID := asn1.ObjectIdentifier{2, 5, 29, 28} // id-ce-issuingDistributionPoint
+	idpe := lints.GetExtWithOID(c.Extensions, idpOID)
+	if idpe == nil {
+		return &lint.LintResult{
+			Status:  lint.Warn,
+			Details: "CRL missing IDP",
+		}
+	}
+
+	// Step inside the outer issuingDistributionPoint sequence to get access to
+	// its constituent fields.
+	idpv := cryptobyte.String(idpe.Value)
+	_ = idpv.ReadASN1(&idpv, cryptobyte_asn1.SEQUENCE)
```

Comment:
Why is the return value ignored here?

Reply:
This operation is also performed in `//linter/lints/cpcps/lint_crl_has_idp.go` and I figured that lint would take care of verifying the issuingDistributionPoint sequence exists and can be read fine. What I'm realizing now is that without checking if the read is unsuccessful here then the lint would still run with the default BRValidity value, but that's a problem.

Reply:
Makes sense! And also, ignoring errors has a secondary problem: if there was a failure to parse the ASN.1 here, there will likely be another failure lower down, and it would be harder to diagnose than if the error is handled at the site it first occurs.